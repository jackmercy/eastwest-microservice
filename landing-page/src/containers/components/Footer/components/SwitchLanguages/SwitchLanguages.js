import React, { useState, memo, useRef, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { useDispatch } from 'react-redux';
import { updateIntl } from 'react-intl-redux';

import intlMessages from '../../../../../intl';
import { storageUtil } from '../../../../../utils';
import { useOnClickOutside } from '../../../../../hooks';
import SectionTitle from '../SectionTitle';

import useStyles from './switchLanguages.styles';

const SwitchLanguages = () => {
  const ref = useRef();
  const intl = useIntl();
  const classes = useStyles();
  const dispatch = useDispatch();
  const languages = useMemo(
    () => ({
      en: {
        key: 'en',
        title: intl.formatMessage({ id: 'LANGUAGE.EN' }),
      },
      // vi: {
      //   key: 'vi',
      //   title: intl.formatMessage({ id: 'LANGUAGE.VI' }),
      // },
    }),
    [],
  );
  const [openMenu, setOpenMenu] = useState(false);

  const toggleMenu = () => {
    // MVP disabled
    // setOpenMenu(!openMenu);
  };

  const handleItemClick = (locale) => {
    if (locale !== intl.locale) {
      storageUtil.setLocale(locale);
      dispatch(
        updateIntl({
          locale,
          messages: intlMessages[locale],
        }),
      );
    }
    setOpenMenu(false);
  };

  useOnClickOutside(ref, () => {
    setOpenMenu(false);
  });

  return (
    <div className={classes.root}>
      <SectionTitle
        title={intl.formatMessage({ id: 'FOOTER.SWITCH_LANGUAGES.TITLE' })}
        dataTest="switch-languages-title"
      />
      <div className={classes.content} ref={ref}>
        <span
          role="presentation"
          onClick={toggleMenu}
          className={classes.toggleButton}
          data-test="switch-languages-toggle-button"
        >
          <i className="icon-global" />
          {languages[intl.locale].title}
          <i className="icon-triangle-down" />
        </span>
        {openMenu ? (
          <ul className={classes.menu} data-test="switch-languages-menu">
            {Object.values(languages).map((item) => (
              <li
                key={`switch-language-${item.key}`}
                role="presentation"
                className={classes.menuItem}
                onClick={() => handleItemClick(item.key)}
              >
                {item.title}
              </li>
            ))}
          </ul>
        ) : null}
      </div>
    </div>
  );
};

export default memo(SwitchLanguages);
