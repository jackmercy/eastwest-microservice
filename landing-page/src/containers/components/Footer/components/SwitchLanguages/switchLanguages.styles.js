import { createUseStyles } from 'react-jss';

import { rem } from '../../../../../utils/jss';
import COLORS from '../../../../../constants/COLORS';
import BREAKPOINTS from '../../../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {},
  content: { marginTop: rem(8), position: 'relative' },
  toggleButton: {
    display: 'flex',
    alignItems: 'center',
    cursor: 'pointer',
    fontSize: rem(16),
    lineHeight: rem(24),
    color: COLORS.textWhite,
    '&:hover [class^=icon]:last-child': { color: COLORS.primary },
    '& [class^=icon]': {
      color: COLORS.textWhite,
      '&:first-child': { fontSize: rem(24), marginRight: rem(8) },
      '&:last-child': { fontSize: rem(7), marginLeft: rem(8) },
    },
  },
  menu: {
    minWidth: rem(180),
    margin: 0,
    zIndex: 999,
    display: 'block',
    position: 'absolute',
    backgroundColor: COLORS.bgWhite,
    boxShadow: '2px 4px 12px rgba(0, 0, 0, 0.15)',
    borderRadius: rem(5),
    padding: [rem(8), 0],
    left: 0,
    top: rem(30),
    listStyleType: 'none',
  },
  menuItem: {
    whiteSpace: 'nowrap',
    padding: [rem(8), rem(16)],
    '&:hover': {
      cursor: 'pointer',
      backgroundColor: 'rgba(211, 211, 211, 0.2)',
    },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { marginTop: rem(32) },
  },
});
