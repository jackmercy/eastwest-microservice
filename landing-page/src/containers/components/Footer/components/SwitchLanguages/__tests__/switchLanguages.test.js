import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SwitchLanguages from '../SwitchLanguages';

describe('Testing Footer > SwitchLanguages', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<SwitchLanguages />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<SwitchLanguages />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="switch-languages-title"]')).toHaveLength(
      1,
    );
  });

  it('should have a toggle button', () => {
    expect(
      wrapper.find('[data-test="switch-languages-toggle-button"]'),
    ).toHaveLength(1);
  });

  // MVP hidden
  // describe('Testing menu', () => {
  //   beforeAll(() => {
  //     wrapper
  //       .find('[data-test="switch-languages-toggle-button"]')
  //       .simulate('click');
  //   });

  //   it('should show menu when click toggle button', () => {
  //     expect(wrapper.find('[data-test="switch-languages-menu"]')).toHaveLength(
  //       1,
  //     );
  //   });

  //   it('menu should support 1 languages', () => {
  //     expect(
  //       wrapper.find('[data-test="switch-languages-menu"]').children(),
  //     ).toHaveLength(1);
  //   });
  // });
});
