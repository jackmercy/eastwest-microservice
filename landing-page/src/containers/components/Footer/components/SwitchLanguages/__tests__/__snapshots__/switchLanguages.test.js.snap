// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer > SwitchLanguages should render match snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <Memo(SectionTitle)
    dataTest="switch-languages-title"
    title="Choose your language"
  />
  <div
    className="content-0-2-2"
  >
    <span
      className="toggleButton-0-2-3"
      data-test="switch-languages-toggle-button"
      onClick={[Function]}
      role="presentation"
    >
      <i
        className="icon-global"
      />
      English
      <i
        className="icon-triangle-down"
      />
    </span>
  </div>
</div>
`;
